storages/backends/base.py

```diff
@@ -0,0 +1,9 @@
+from django.core.files.base import File
+from django.core.files.storage import Storage
+
+class BaseFile(File):
```

Comment:
@jdufresne It doesn't provide hooks at this moment as this is just a skeleton proposal, but it will certainly have hooks if we agree that this is a good approach.